### PR TITLE
fix(rollapp): restore the ability to import rollapp state

### LINKED ISSUE #880: Restore the ability to import rollapp state
previously we've blocked it as we weren't sure about the fine details of rollapp behavior upon hub forking.
After the research done [here](https://www.notion.so/dymension/Research-on-Dymension-Forking-42dc33659a4c466e964510e510862b9f?pvs=4), we can now import the relevant parts as suggested in the research and remove the enforcment that only params can be imported.

### PR DESCRIPTION
**Pull request by anhductn2001**
## Description

----

Closes #880 

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----;

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: x/delayedack/genesis.go [hunk 1]
```diff
@@ -9,9 +9,8 @@ import (
 // InitGenesis initializes the module's state from a provided genesis state.
 func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
 	k.SetParams(ctx, genState.Params)
-	// Validate all other genesis fields are empty
-	if len(genState.RollappPackets) != 0 {
-		panic("Only params can be initialized at genesis")
+	for _, packet := range genState.RollappPackets {
+		k.SetRollappPacket(ctx, packet)
```
Comment (srene) [inline x/delayedack/genesis.go:13]: i think we need to re-add the tests for init and export genesis for the delayedack module

**Review by srene — COMMENTED** [review]
i thi

File: x/delayedack/genesis.go (hunk 1 above)
Comment (anhductn2001) [reply x/delayedack/genesis.go:13]: I see that we don't have genesis_test for delayedack module before.

Comment (anhductn2001) [reply x/delayedack/genesis.go:13]: cc: @srene

**Review by srene — APPROVED** [review]

**Comment by anhductn2001** [comment]
Ok, need 1 more review to merge it @omritoptix

File: x/sequencer/types/genesis.go [hunk 2]
```diff
@@ -11,5 +13,21 @@ func DefaultGenesis() *GenesisState {
...
+	// Check for duplicated index in sequencer
+	sequencerIndexMap := make(map[string]struct{})
+
+	for _, elem := range gs.SequencerList {
+
+		// FIXME: should run validation on the sequencer objects
+
+		index := string(SequencerKey(elem.SequencerAddress))
+		if _, ok := sequencerIndexMap[index]; ok {
+			return fmt.Errorf("duplicated index for sequencer")
```
Comment (zale144) [inline x/sequencer/types/genesis.go:25]: predefine an error: `ErrDuplicateSequencerIndex = errorsmod.Register(ModuleName, 1019, "duplicate index for sequencer")` in errors.go

**Review by zale144 — CHANGES_REQUESTED** [review]

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**